src/main/java/org/interledger/Condition.java

```diff
@@ -0,0 +1,66 @@
+package org.interledger;
+
+import java.util.Arrays;
+
+/**
+ * The execution condition attached to all transfers in an Interledger payment.
+ */
+public class Condition {
+
+  private byte[] hash;
+
+  /**
+   * Create a new {@code Condition} from a SHA-256 hash.
+   * 
+   * @param hash The SHA-256 hash of a preimage that is used as the fulfillment of this condition
+   */
+  public Condition(byte[] hash) {
+
+    if (hash == null || hash.length != 32) {
+      throw new IllegalArgumentException("Hash must be 32 bytes.");
+    }
+
+    this.hash = new byte[32];
+    System.arraycopy(hash, 0, this.hash, 0, 32);
```

Comment:
We should prefer the following style for conciseness and portability:

`this.hash = Arrays.copyOf(hash, hash.length);`